includes/class-reader-activation.php

```diff
@@ -257,6 +292,327 @@ public static function auth_cookie_expiration( $length, $user_id, $remember ) {
+	 * @return string Account link HTML or empty string.
+	 */
+	private static function get_account_link() {
+		$account_url = '';
+		if ( function_exists( 'wc_get_account_endpoint_url' ) ) {
+			$account_url = \wc_get_account_endpoint_url( 'dashboard' );
+		}
+
+		/** Do not render link for authenticated readers if account page doesn't exist. */
+		if ( empty( $account_url ) && \is_user_logged_in() ) {
+			return '';
+		}
+
+		$labels = [
+			'signedin'  => \__( 'Account', 'newspack' ),
+			'signedout' => \__( 'Sign In', 'newspack' ),
+		];
+		$label  = \is_user_logged_in() ? 'signedin' : 'signedout';
+
+		$link  = '<span class="newspack-reader__account-link" data-labels="' . \esc_attr( htmlspecialchars( \wp_json_encode( $labels ), ENT_QUOTES, 'UTF-8' ) ) . '">';
+		$link .= '<a href="' . \esc_url_raw( $account_url ?? '#' ) . '" data-newspack-reader-account-link>';
+		$link .= '<span class="newspack-reader__account-link__icon">';
+		$link .= self::get_account_icon();
+		$link .= '</span>';
+		$link .= '<span class="newspack-reader__account-link__label">' . \esc_html( $labels[ $label ] ) . '</span>';
+		$link .= '</a>';
+		$link .= '</span>';
+
+		/**
+		 * Filters the HTML for the reader account link.
+		 *
+		 * @param string $link HTML for the reader account link.
+		 */
+		return apply_filters( 'newspack_reader_account_link', $link );
+	}
+
+	/**
+	 * Renders reader authentication form
+	 */
+	public static function render_auth_form() {
+		if ( \is_user_logged_in() ) {
+			return;
+		}
+		$message    = '';
+		$classnames = [ 'newspack-reader__auth-form' ];
+		// phpcs:disable WordPress.Security.NonceVerification.Recommended
+		if ( isset( $_GET['reader_authenticated'] ) && isset( $_GET['message'] ) ) {
+			$message      = \sanitize_text_field( $_GET['message'] );
+			$classnames[] = 'visible';
+		}
+		// phpcs:enable
+		?>
+		<div id="newspack-reader-auth" class="<?php echo \esc_attr( implode( ' ', $classnames ) ); ?>">
+			<div class="newspack-reader__auth-form__wrapper">
+				<button class="newspack-reader__auth-form__close" data-close aria-label="<?php \esc_attr_e( 'Close Authentication Form', 'newspack' ); ?>">
+					<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" role="img" aria-hidden="true" focusable="false">
+						<path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/>
+					</svg>
+				</button>
+				<div class="newspack-reader__auth-form__content">
+					<form method="post" target="_top">
+						<input type="hidden" name="<?php echo \esc_attr( self::AUTH_FORM_ACTION ); ?>" value="1" />
+						<input type="hidden" name="action" value="link" />
+						<div class="newspack-reader__auth-form__header">
+							<h2><?php _e( 'Sign In', 'newspack' ); ?></h2>
+							<a href="#" class="form-actions action-item action-pwd action-link register-link" data-set-action="register"><?php \esc_html_e( "I don't have an account", 'newspack' ); ?></a>
```

Comment:
In the register state (for readers with no account), this link should not be visible, and the header should say "Register" instead of "Sign In". Current state:

<img width="478" alt="image" src="https://user-images.githubusercontent.com/7383192/179950363-0860e575-23e8-4318-a619-a44acecb06e1.png">

Reply:
c151def05cfc60e00e7ca8dddccb9c141baa235b introduces dynamic labels. I'm using "Sign Up" as it is the proposed label on the design. It also adds the "I already have an account" toggle while in the "Sign Up" state.

Reply:
@miguelpeixe This was not addressed in the latest changes 👆

Reply:
Which part wasn't addressed?

Reply:
Wait, I'm having some GH sync issues…

Reply:
All good here!